app/src/main/java/com/github/pockethub/android/ui/notification/NotificationListFragment.java

```diff
@@ -0,0 +1,181 @@
+package com.github.pockethub.android.ui.notification;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.github.pockethub.android.R;
+import com.github.pockethub.android.core.issue.IssueUriMatcher;
+import com.github.pockethub.android.ui.DialogFragment;
+import com.github.pockethub.android.ui.issue.IssuesViewActivity;
+import com.github.pockethub.android.util.ToastUtils;
+import com.meisolsson.githubsdk.core.ServiceGenerator;
+import com.meisolsson.githubsdk.model.Issue;
+import com.meisolsson.githubsdk.model.NotificationThread;
+import com.meisolsson.githubsdk.model.Page;
+import com.meisolsson.githubsdk.model.Repository;
+import com.meisolsson.githubsdk.model.request.NotificationReadRequest;
+import com.meisolsson.githubsdk.service.activity.NotificationService;
+
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NotificationListFragment extends DialogFragment
+        implements NotificationReadListener, AdapterView.OnItemClickListener {
+
+    /**
+     * Filters for the request to GitHub.
+     */
+    private Map<String, Object> filters = new HashMap<>();
+
+    private NotificationListAdapter adapter;
+
+    private ListView list;
+    private TextView emptyText;
+    private ProgressBar progressBar;
+    private SwipeRefreshLayout refreshLayout;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Bundle args = getArguments();
+        if (args.containsKey("filter")) {
```

Comment:
How about make “filter” to a constant(KEY_FILTER) in NotificationListFragment, since it's an argument key and many place need to use it. typo it may cause bug in other place.